app/SearchApi/BcGov.Fams3.Redis/ICacheService.cs

```diff
@@ -116,13 +116,22 @@ public async Task Save(string key, dynamic data, TimeSpan expiry)
 
         public async Task<string> Get(string key)
         {
-
             if (string.IsNullOrEmpty(key)) throw new ArgumentNullException("Get : Key cannot be null");
 
-            string strData = await _stackRedisCacheClient.Db0.GetAsync<string>(key);
-
-            return strData;
-
+            for (int attempt = 0; attempt < 3; attempt++)
+            {
+                try
+                {
+                    string strData = await _stackRedisCacheClient.Db0.GetAsync<string>(key);
+                    return strData;
+                }
+                catch (Exception e)
+                {
+                    if (attempt == 2) throw e;
+                    Thread.Sleep(50);
+                }

```

Comment:
Will this not increase bottleneck.. shouldn't we look outside the library ?

Reply:
it should not increase as in 99%, the first attempt will be successful, then it returns. 
I added things here because I am afraid there will be other places also have this problem, but we did not know.
On the other hand, adding to outside lib is safe ( we only deal with the issue we found).
So, I will move it to outside the lib.